src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -1236,7 +1245,9 @@ void primitive_inst::load(cldnn::BinaryInputBuffer& ib) {
             _outputs[0] = get_network().get_engine().allocate_memory(output_layout, _allocation_type);
         }
     } else {
-        if ((!can_share_buffer()) || can_be_optimized() || is_output()) {
+        if (memory_reuse_by_user == false) {
+            _outputs[0] = get_network().get_memory_pool().get_memory(output_layout, id(), get_network_id(), _node_mem_deps, _allocation_type, false);
+        } else if ((!can_share_buffer()) || can_be_optimized() || is_output()) {
             _outputs[0] = get_network().get_engine().allocate_memory(output_layout, _allocation_type);
         } else {
             _outputs[0] = get_network().get_memory_pool().get_memory(output_layout, id(), get_network_id(), _node_mem_deps, _allocation_type, true);
```

Comment:
How about more simply, 

```
if ((!can_share_buffer()) || can_be_optimized() || is_output()) {
 ...
} else {
  _outputs[0] = get_network().get_memory_pool().get_memory(output_layout, id(), get_network_id(), _node_mem_Deps, _allocation_type, can_reuse_memory);
}
```
}

Reply:
I updated the code as you reviewed. Thank you.